backend/http/firehose.go

```diff
@@ -0,0 +1,68 @@
+package http
+
+import (
+	"compress/gzip"
+	"encoding/base64"
+	"encoding/json"
+	"github.com/google/uuid"
+	model2 "github.com/highlight-run/highlight/backend/model"
+	"io"
+	"net/http"
+	"strings"
+)
+
+type FirehosePayload struct {
+	RequestId string
+	Timestamp int64
+	Records   []struct {
+		Data string
+	}
+}
+
+func ExtractFirehoseMetadata(r *http.Request, body []byte) (int, *FirehosePayload, [][]byte, error) {
```

Comment:
NB reviewer: this was done to share the firehose specific logic with OTLP metrics ingest, but the `otel.go` change did not end up using this as we will use the `awsfirehose` collector receiver going forward. the `logging.go` firehose specific endpoint will be kept for backwards compat. but i'll update the docs to start using the firehose receiver for all firehose data